### ISSUE TITLE
Method signature compatibility

### ISSUE THREAD

**Issue by mudrd8mz** [wiris/moodle-qtype_multianswerwiris#2]
When previewing the question on Moodle 3.1, I am seeing debugging warning:
```
Warning: Declaration of
qtype_multianswerwiris_helper_renderer::subquestion(question_attempt $qa, question_display_options $options, $index, $subq)
should be compatible with
qtype_multianswer_renderer::subquestion(question_attempt $qa, question_display_options $options, $index, question_graded_automatically $subq)
in .../question/type/multianswerwiris/renderer.php on line 27
```

**Comment by mudrd8mz** [comment]
I see that `qtype_multianswerwiris_question::get_renderer()` has a related comment and tries to hide this warning by setting the error reporting.

**Comment by manuelwiris** [comment]
Hi David, 

PHP 7.0 removes E_STRICT error levels ([](http://php.net/manual/en/migration70.incompatible.php)) and method signature compatibility has been reclassified as a E_WARNING level.

We are try to hiding E_STRICT errors because WIRIS questions extends `question_with_responses `instead of `question_graded_automatically`.

To fix this we can hide E_WARNING (for PHP 7.x)  errors on `multianswerwiris/renderer.php` file when `qtype_multianswerwiris_helper_render `is loaded and restore to the default value once the class is loaded. 

With this change, if is a E_WARNING during function execution it will be caught.

**Comment by mudrd8mz** [comment]
> WIRIS questions extends question_with_responses instead of question_graded_automatically

Yes, that is the root of this issue. I am wondering why. If the parent renderer really requires `question_graded_automatically` then maybe there is a reason. In which case not providing the expected types could lead to serious problems.

I generally believe that hiding warnings like this is a bad thing to do.
